### DISCUSSION TITLE
Crashes

### DISCUSSION THREAD

**Discussion by 404User777** [SudhanPlayz/Discord-MusicBot#688] [Q&A] [answered]
The bot crashes when it turns out to be under other discord roles. For example after doing .stats the bot fails to send the message despite all its permissions and crashes. Here is the error:
```

DiscordAPIError: Missing Permissions
    at RequestHandler.execute (/root/Music/node_modules/discord.js/src/rest/RequestHandler.js:154:13)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async RequestHandler.push (/root/Music/node_modules/discord.js/src/rest/RequestHandler.js:39:14) {
  method: 'post',
  path: '/channels/93631298511319869/messages',
  code: 50013,
  httpStatus: 403
}
```

**Comment by DarrenOfficial** [comment]
Please put the steps on how to reproduce this bug

**Comment by kotru21** [comment]
As Darren said,provide more info about this crash :)
It may happen because of wrong bot's (invite) permissions or not(or wrong) setted gateway intents.

**Comment by 404User777** [comment]
The bot has the default permissions already present in the code and on the discord developer portal, all 3 intent options are active. To reproduce the bug just put the role of the bot under all and run the command .stats . I don't know how to explain since it doesn't happen in all servers but only in some. It seems that without a .catch the bot crashes

**Comment by kotru21** [comment]
> The bot has the default permissions already present in the code and on the discord developer portal, all 3 intent options are active. To reproduce the bug just put the role of the bot under all and run the command .stats . I don't know how to explain since it doesn't happen in all servers but only in some. It seems that without a .catch the bot crashes

Maybe someone invited bot with invalid permissions?
I mean that when you invite bot to yor server you need to put correct perms to your invite link
 https://discord.com/oauth2/authorize?client_id=INSERT_CLIENT_ID_HERE&scope=bot&permissions=PUTHERECORRECTPERMISSIONS
To fix that issue,try reinviting bot with correct perms:)

**Comment by t0g3pii** [comment, marked as answer]
The Problem is the following:

instead of using &scope=bot you need to use
&scope=applications.commands%20bot

The Problem here is that the Script tries to add / (Slash-Commands) which it doesnt have perms for. The Scope is missing on the URL.

**Comment by t0g3pii** [comment]
https://discord.com/oauth2/authorize?client_id=INSERT_CLIENT_ID_HERE&&scope=applications.commands%20bot&permissions=PUTHERECORRECTPERMISSIONS

would be the correct link
